gtars/src/refget/store.rs

```diff
@@ -250,6 +262,253 @@ impl GlobalRefgetStore {
+                    }
+                };
+
+            // Handle the -1 return for parsing errors in parse_bedlike_file
+
+            if parsed_start == -1 || parsed_end == -1 {
+                eprintln!(
+                    "Warning: Skipping line {} due to invalid start or end coordinates: '{}'",
+                    line_num + 1,
+                    line_string
+                );
+                continue;
+            }
+
+
+            if previous_parsed_chr != parsed_chr {
+                previous_parsed_chr = parsed_chr.clone();
+
+                let result = match self.get_sequence_by_collection_and_name(&collection_digest, &*parsed_chr) {
+                    Some(seq_record) => seq_record,
+                    None => {
+                        eprintln!(
+                            "Warning: Skipping line {} because sequence '{}' not found in collection '{}'.",
+                            line_num + 1,
+                            parsed_chr,
+                            String::from_utf8_lossy(collection_digest.as_ref())
+                        );
+                        continue;
+                    }
+                };
+
+                current_seq_digest = result.metadata.sha512t24u.clone();
+                current_header = format!(
+                    ">{} {} {} {} {}",
+                    result.metadata.name,
+                    result.metadata.length,
+                    result.metadata.alphabet,
+                    result.metadata.sha512t24u,
+                    result.metadata.md5
+                );
+            }
+
+
+
+            let retrieved_substring = match self.get_substring(&current_seq_digest, parsed_start as usize, parsed_end as usize) {
+                Some(substring) => substring,
+                None => {
+                    eprintln!(
+                        "Warning: Skipping line {} because substring for digest '{}' from {} to {} not found or invalid.",
+                        line_num + 1,
+                        current_seq_digest, parsed_start, parsed_end
+                    );
+                    continue;
+                }
+            };
+            if previous_header != current_header {
+                let prefix = if previous_header.is_empty() { "" } else { "\n" };
+
+                previous_header = current_header.clone();
+
+                // Combine the prefix, current_header, and a trailing newline
+                let header_to_be_written = format!("{}{}\n", prefix, current_header);
+                output_file.write_all(header_to_be_written.as_bytes())?;
+            }
+
+            output_file.write_all(retrieved_substring.as_ref())?;
+        }
+
+        Ok(())
+    }
+
+    /// Given a Sequence Collection digest and a bed file path,
+    /// retrieve sequences and return them as a Vec<RetrievedSequence>.
+    /// Lines that cannot be parsed or sequences that cannot be retrieved will be skipped
+    pub fn get_seqs_bed_file_to_vec<K: AsRef<[u8]>>(
```

Comment:
Seems like a lot of duplicated functionality. Maybe we could abstract that out and then return an iterator lover the sequence collection and/or bedfile, then one can call it and dump to file or collect into `Vec`